lib/Newsletter/AutomatedLatestContent.php

```diff
@@ -84,6 +95,7 @@ function getPosts($args, $posts_to_exclude = array()) {
     $posts = WPFunctions::get()->getPosts($parameters);
     $this->logPosts($posts);
 
+    WPFunctions::get()->removeAction('pre_get_posts', [$this, 'ensureConstistentQueryType'], $filter_priority);
```

Comment:
I am not sure how WP compares the callback functions. if it doesn't do a deep comparison, would the two `[$this, 'ensureConsistentQueryType']` be equal?

Reply:
It should be OK. In https://codex.wordpress.org/Function_Reference/remove_action they mention:

> If an action has been added from within a class, for example by a plugin, removing it will require accessing the class through a variable that holds the class instance.

> ```
> global $my_class;
> remove_action( 'wp_footer', array( $my_class, 'class_function_being_removed' ) );
> ```